code/modules/xenoarchaeology/traits/activators.dm

```diff
@@ -0,0 +1,664 @@
+	trigger_artifact(target, XENOA_ACTIVATION_TOUCH)
+
+//Credit variant
+/datum/xenoartifact_trait/activator/item_key/greedy/credit
+	label_name = "Greedy Δ"
+	label_desc = "Greedy Δ: The artifact seems to be made of a collective material. This material seems to be triggered by inserting credit holochips."
+	key_item = /obj/item/holochip
+	conductivity = 8
+	///How many credits we need to activate
+	var/credit_requirement = 1
+
+/datum/xenoartifact_trait/activator/item_key/greedy/credit/get_dictionary_hint()
+	. = ..()
+	return list(XENOA_TRAIT_HINT_TRIGGER("credit holochip"), XENOA_TRAIT_HINT_TWIN, XENOA_TRAIT_HINT_TWIN_VARIANT("accept credit holochips to activate"))
+
+/datum/xenoartifact_trait/activator/item_key/greedy/credit/handle_input(atom/item, atom/target)
+	var/obj/item/holochip/C = item
+	if(C.credits < credit_requirement)
+		to_chat(target, "<span class='warning'>[parent.parent] demands more than your meager offering!</span>")
+		playsound(parent.parent, 'sound/machines/buzz-sigh.ogg', 50, TRUE)
+		return
+	C.forceMove(parent.parent)
+	playsound(parent.parent, 'sound/machines/terminal_insert_disc.ogg', 50, TRUE)
+	trigger_artifact(target, XENOA_ACTIVATION_TOUCH)
+	credit_requirement += 1
+
+/*
+	Weighted
+	This trait activates the artifact when it is picked up
+*/
+/datum/xenoartifact_trait/activator/weighted
+	label_name = "Weighted"
+	label_desc = "Weighted: The artifact seems to be made of a weighted material. This material seems to be triggered by motion, such as being picked up."
+	flags = XENOA_BLUESPACE_TRAIT | XENOA_URANIUM_TRAIT | XENOA_BANANIUM_TRAIT | XENOA_PEARL_TRAIT
+	blacklist_traits = list(/datum/xenoartifact_trait/minor/dense)
+	weight = 32
+	incompatabilities = TRAIT_INCOMPATIBLE_MOB | TRAIT_INCOMPATIBLE_STRUCTURE
+
+/datum/xenoartifact_trait/activator/weighted/New(atom/_parent)
+	. = ..()
+	if(!parent?.parent)
+		return
+	RegisterSignal(parent?.parent, COMSIG_ITEM_EQUIPPED, TYPE_PROC_REF(/datum/xenoartifact_trait/activator, translation_type_d))
+
+/datum/xenoartifact_trait/activator/weighted/translation_type_d(datum/source, atom/target)
+	trigger_artifact(target, XENOA_ACTIVATION_TOUCH)
+
+/*
+	Pitched
+	This trait activates the artifact when it is thrown
+*/
+/datum/xenoartifact_trait/activator/pitched
+	label_name = "Pitched"
+	label_desc = "Pitched: The artifact seems to be made of an aerodynamic material. This material seems to be triggered by motion, such as being thrown."
+	flags = XENOA_BLUESPACE_TRAIT | XENOA_URANIUM_TRAIT | XENOA_BANANIUM_TRAIT | XENOA_PEARL_TRAIT
+	blacklist_traits = list(/datum/xenoartifact_trait/minor/dense)
+	weight = -8
+	incompatabilities = TRAIT_INCOMPATIBLE_MOB | TRAIT_INCOMPATIBLE_STRUCTURE
+
+/datum/xenoartifact_trait/activator/pitched/New(atom/_parent)
+	. = ..()
+	if(!parent?.parent)
+		return
+	RegisterSignal(parent?.parent, COMSIG_MOVABLE_IMPACT, TYPE_PROC_REF(/datum/xenoartifact_trait/activator, translation_type_a))
```

Comment:
There are so many parents doing so many different things, being so many different types. This code is extremely difficult to read. Is the new parent different from the component parent?

Reply:
![image](https://github.com/user-attachments/assets/b122f1d5-c617-43ea-ad5d-170f1528b53c)
![image](https://github.com/user-attachments/assets/3cf6fce0-1895-476c-8d2f-f880afc124d2)
What's the point of commenting?

Reply:
Because the code overall is extremely difficult to read in this case during review due to a combination of confusing variable names and unclear logic. _parent is an atom, parent is a component and parent.parent is a datum, all representing potentially different things. It is unclear why _parent and parent.parent need to be seperate things and the benefit of the trait being directly attached to a component instead of being a singleton that acts upon a passed atom is unclear.